sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -556,6 +556,98 @@ Parameters:
 
 |===
 
+==== Dynamic Command Groups
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+class dynamic_command_group {
+public:
+  dynamic_command_group(
+      command_graph<graph_state::modifiable> graph, std::vector<std::function<void(handler &)>> cgfList);
+
+  void set_active_cgf(int cgfIndex);
+};
+----
+
+Dynamic command-groups can be added as nodes to a graph. They provide a mechanism that
+allows updating the command-group function of a node after the graph is finalized.
+There is always one command-group function in the dynamic command-group that is set
+as active. When a dynamic command-group node is executed, the kernel of the active
+command-group function will be run and all the other command-group functions in
+`cgfList` will be ignored.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating command-groups.
+
+===== Limitations:
+
+Dynamic command-groups can only be used to update kernels. Trying to update a command group
+function that contains memory operations will result in an error.
+
+All the command-group functions in a dynamic command group must have identical dependencies.
+It is not allowed for a dynamic command group to have command-group functions that would
+result in a change to the graph topology when set to active. In practice, this means that
+any calls to `handler.depends_on()` must be identical for all the command-group functions
+in a dynamic command-group.
+
+Table {counter: tableNumber}. Member functions of the `dynamic_command_group` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_command_group(command_graph<graph_state::modifiable> graph, std::vector<std::function<void(handler &)>> cgfList);
```

Comment:
The length of this signature means it doesn't read well when rendered. Would split it up like

```
dynamic_command_group(command_graph<graph_state::modifiable> graph,
                      std::vector<std::function<void(handler &)>> cgfList);
```